crates/core/tedge_agent/src/operation_workflows/persist.rs

```diff
@@ -74,7 +71,7 @@ impl WorkflowRepository {
         // First, all the user-defined workflows are loaded
         let dir_path = &self.custom_workflows_dir.clone();
         if let Err(err) = self
-            .load_operation_workflows(WorkflowSource::UserDefined, dir_path)
+            .load_operation_workflows(WorkflowSource::UserDefined(dir_path))
```

Comment:
Opinion: Although I understand that `WorkflowSource` being a generic type allows such generic usages, but this one with the directory path seems slightly overloaded. I agree that we can still defend it since the directory roots are also linked to the source type.